src/pages/dataTableUtilities/getIndicatorTableData.js

```diff
@@ -44,81 +76,90 @@ const COLUMNS = {
  * @param {Period} period
  */
 const initialiseRowColumnValue = (row, column, period) => {
-  UIDatabase.write(() => {
-    createRecord(UIDatabase, 'IndicatorValue', row, column, period);
-  });
+  UIDatabase.write(() => createRecord(UIDatabase, 'IndicatorValue', row, column, period));
+  return UIDatabase.objects('IndicatorValue')
+    .filtered('row == $0 AND column == $1 AND period == $2', row, column, period);
```

Comment:
```
let indicator;
UIDatabase.write(() => { indicator = createRecord(UIDatabase, 'IndicatorValue', row, column, period) });
return indicator;
```

ugly but 🤷‍♂

Reply:
Feels wrong but is a lot better performance-wise. 425a8d2d.